src/service.rs

```diff
@@ -424,14 +836,410 @@ impl Service {
                         self.send_ping(enr);
                     }
                 }
+                Some(Ok((active_topic, active_ticket))) = self.tickets.next() => {
+                    // When the ticket time expires a new REGTOPIC request is automatically sent to the
+                    // ticket issuer and the registration attempt stays in the [`RegistrationState::Ticket`]
+                    // from sending the first REGTOPIC request to this contact for this topic.
+                    self.reg_topic_request(active_ticket.contact(), active_topic.topic().clone(), RequestTicket::RemotelyIssued(active_ticket.ticket()));
+                }
+                Some(topic_query_progress) = self.active_topic_queries.next() => {
+                    match topic_query_progress {
+                        TopicQueryState::Finished(topic_hash) | TopicQueryState::TimedOut(topic_hash) => {
+                            if let Some(query) = self.active_topic_queries.queries.remove(&topic_hash) {
+                                if let Some(callback) = query.callback {
+                                    if callback.send(Ok(query.results.into_values().collect::<Vec<_>>())).is_err() {
```

Comment:
I found three instances of usage of the callback and all are ok responses. So it's probably good to remove the result from the sent type